File: 5/5-299.py
def r(n):
    n_bi = bin(n)[2:]
    for _ in range(2):
        c1 = n_bi.count('1')
        if c1 % 2:
            n_bi = '1' + n_bi + '00'
        else:
            n_bi = n_bi[1:].lstrip('0')
    return int(n_bi, 2)


n = 1
while r(n) <= 100:
    n += 1
print(n)

File: Dima/5/5-299.pas
// Автор В.Н. Шубинкин

uses School;

function R(N: integer): integer;
begin
  var N_bi := Bin(N);
  loop 2 do
  begin
    var c1 := N_bi.CountOf('1');
    if c1 mod 2 = 0 then N_bi := N_bi[2:].ToInteger.ToString
    else N_bi := 1 + N_bi + '00'
  end;
  result := Dec(N_bi, 2)
end;


begin
  var n := 1;
  while R(n) <= 100 do n += 1;
  print(n)
end.